### ISSUE TITLE
Data Retrieval issues 

### ISSUE THREAD

**Issue by mattiasghodsian** [benrr101/node-taglib-sharp#104]
Am trying to figure out how to read data from flac and mp3 files., encountering undefined fields such as 'date'. What am i missing here ? all data comes from MusicBrainz

Would be nice to have a 'tags' function for retrieving all available data as object.

![image](https://github.com/benrr101/node-taglib-sharp/assets/20265517/f9bd1002-ed2b-4697-b92b-0575b3d763f6)
![image](https://github.com/benrr101/node-taglib-sharp/assets/20265517/3b0b2297-2379-4858-a464-4ee70aadae93)

```js
 const myFile = File.createFromPath(filePath); // flac or mp3

 console.log(myFile.tag.title);
 console.log(myFile.tag.date)
 console.log(myFile.tag.artists);
  ```

**Comment by Ed1ks** [comment]
+1 
please add date tag

**Comment by benrr101** [comment]
Hi @mattiasghodsian sorry for the super long delay in getting back to you one this one. No good excuses here, just other projects and life getting in the way.

For your first question, regarding undefined fields, I think there's a bit of a misunderstanding. Each application/library that can read tags will have variations between mapping. It's important to note that there are different tagging formats that different file types can use (see https://github.com/benrr101/node-taglib-sharp/wiki/02-Tagging-Support#the-table) and each tagging format has its own fields. The `Tag` class makes a best effort to use de facto mappings between commonly used tags and the underlying tag fields. See https://github.com/benrr101/node-taglib-sharp/blob/develop/docs/classes/Tag.md for a list of the fields that are available.

For your second question, which sort of boils down to "where are `artists` and `date`?"
* `artists` are stored as `performers` in the `Tag` class
* `date` is only partially supported in the unified `Tag` class. Only some tag formats support a free-form text date field, while some support just the year. To cover this lowest common denominator, you'll have to use `year` on the `Tag` class.

Keep in mind, each application may have a slightly different mapping for a field to the underlying tag construct. `Tag` is best effort to provide a unified experience, but it will never cover all fields or mappings. If you want to do something outside of what `Tag` offers, you can always write directly to the underlying tag in an advanced way. Eg, directly accessing frames in `Id3v2Tag` or directly accessing fields in `XiphComment` classes. I've been meaning to properly write this out in the wiki, but at a certain point I just start duplicating the documentation for the formats...

As for exposing all fields as an object, this might not be as simple as you'd like. Like I said before, each tagging format stores fields slightly differently, and it doesn't always lend itself to being a dictionary of fields to values. For instance, ID3v2 uses a collection of "frames" for each field it stores, and each frame has a 4 byte frame type. This would end up giving a dictionary like `{TIT2: 'track title', TALB: 'album name'}`. Xiph comments uses a collection of fields and values which would lend itself to being a dictionary. WebM/Matroska uses a super convoluted collection of nested tags and levels which I honestly have no clue how it could be flattened adequately.

Basically, this is a common issue - each tagging format is different and any attempt to unify access to them will be imperfect. The goal of node-taglib-sharp is to make a best effort at unifying access but offering an escape hatch to manipulate tags on a per-format basis.
